Platform/UTMJailbreak.c

```diff
@@ -32,33 +36,62 @@ kern_return_t catch_exception_raise(mach_port_t exception_port,
                                     exception_type_t exception,
                                     exception_data_t code,
                                     mach_msg_type_number_t code_count) {
-    NSLog(@"Caught exception %d (this should be EXC_SOFTWARE), with code 0x%x (this should be EXC_SOFT_SIGNAL) and subcode %d. Forcing suicide.", exception, *code, code[1]);
+    fprintf(stderr, "Caught exception %d (this should be EXC_SOFTWARE), with code 0x%x (this should be EXC_SOFT_SIGNAL) and subcode %d. Forcing suicide.", exception, *code, code[1]);
     // _exit doesn't seem to work, but this does. ¯\_(ツ)_/¯
     return KERN_FAILURE;
 }
 
-void *exception_handler(void *argument) {
+static void *exception_handler(void *argument) {
     mach_port_t port = *(mach_port_t *)argument;
     mach_msg_server(exc_server, 2048, port, 0);
     return NULL;
 }
 
-static BOOL am_i_being_debugged() {
+static bool am_i_being_debugged() {
     int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_PID, getpid()};
     struct kinfo_proc info = {};
     size_t size = sizeof(info);
     return !sysctl(mib, sizeof(mib) / sizeof(*mib), &info, &size, NULL, 0) && !!(info.kp_proc.p_flag & P_TRACED);
 }
 
-int main(int argc, char * argv[]) {
-    BOOL debugged = am_i_being_debugged();
+
+bool jb_has_jit_entitlement(void) {
+    void *addr = mmap(NULL, PAGE_SIZE, PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON | MAP_PRIVATE | MAP_JIT, -1, 0);
+    if (addr != NULL) {
+        munmap(addr, PAGE_SIZE);
+        return true;
+    } else {
+        return false;
+    }
+}
+
+bool jb_has_ptrace_hack(void) {
+#if defined(NO_PTRACE_HACK)
+    return false;
+#else
+    int res = ptrace(-1, -1, NULL, 0);
+    if (res < 0 && errno == EINVAL) {
+        return true;
+    } else {
+        return false;
+    }
```

Comment:
`return ptrace(-1, -1, NULL, 0) < 0 && errno == EINVAL`?

Reply:
I don’t like having side effects from within if-statements.

Reply:
I was suggesting removing the if statement entirely, since you're returning the condition you're checking for.